src/utils.ts

```diff
@@ -456,3 +456,122 @@ export function setCrypto(c: Object) {
 export function getCrypto(): Object {
     return crypto;
 }
+
+// String averaging based upon https://stackoverflow.com/a/2510816
+// Dev note: We make the alphabet a string because it's easier to write syntactically
+// than arrays. Thankfully, strings implement the useful parts of the Array interface
+// anyhow.
+
+/**
+ * The default alphabet used by string averaging in this SDK. This matches
+ * all usefully printable ASCII characters (0x20-0x7E, inclusive).
+ */
+export const DEFAULT_ALPHABET = [""].reduce(() => {
+    let str = "";
+    for (let c = 0x20; c <= 0x7E; c++) {
+        str += String.fromCharCode(c);
+    }
+    return str;
+}, "");
+
+/**
+ * Pads a string using the given alphabet as a base. The returned string will be
+ * padded at the end with the first character in the alphabet.
+ *
+ * This is intended for use with string averaging.
+ * @param {string} s The string to pad.
+ * @param {number} n The length to pad to.
+ * @param {string} alphabet The alphabet to use as a single string.
+ * @returns {string} The padded string.
+ */
+export function alphabetPad(s: string, n: number, alphabet = DEFAULT_ALPHABET): string {
+    return s.padEnd(n, alphabet[0]);
+}
+
+/**
+ * Converts a baseN number to a string, where N is the alphabet's length.
+ *
+ * This is intended for use with string averaging.
+ * @param {number} n The baseN number.
+ * @param {string} alphabet The alphabet to use as a single string.
+ * @returns {string} The baseN number encoded as a string from the alphabet.
+ */
+export function baseToString(n: number, alphabet = DEFAULT_ALPHABET): string {
+    const len = alphabet.length;
+    if (n < len) {
+        return alphabet[n];
+    }
+    return baseToString(Math.floor(n / len), alphabet) + alphabet[n % len];
+}
+
+/**
+ * Converts a string to a baseN number, where N is the alphabet's length.
+ *
+ * This is intended for use with string averaging.
+ * @param {string} s The string to convert to a number.
+ * @param {string} alphabet The alphabet to use as a single string.
+ * @returns {number} The baseN number.
+ */
+export function stringToBase(s: string, alphabet = DEFAULT_ALPHABET): number {
+    const len = alphabet.length;
+    const reversedStr = Array.from(s).reverse().join(""); // keep as string
```

Comment:
mostly because the math is already complex enough to understand: trying to increase density isn't exactly a good thing, and in practice this is under 1ms for the most complex strings. I'm not too concerned about improving it.

Reply:
gah, but I guess this is outside the range of "unstable, your fault for using" :/

Reply:
if this bit is meant to be unstable then can it be named as such or live in an `unstableUtils.ts` or something? Other unstable things mention they instability